src/import_edf.jl

```diff
@@ -338,6 +390,30 @@ function validate_arrow_prefix(prefix)
     return prefix
 end
 
+# the same EDF.Signal was extracted into multiple `Onda.SampleInfos`.
+struct AmbiguousChannelError <: Exception
+    summary
+end
+
+# return a list of tuples describing sample info channels that are ambiguous
+# because they were extracted from the same edf_signal
+# modifies `infos` and associated `edf_signals::Vector{EDF.Signal}` in place to remove ambiguous signals
+function _ambiguous_channels!(extracted, edf_signal)
+    ambiguous = []
+    for (info, edf_signals) in extracted
+        i = findfirst(==(edf_signal), edf_signals)
+        isnothing(i) && continue
+        channel_name = info.channels[i]
+        summary = (info.kind, channel_name, info.sample_unit)
+        push!(ambiguous, (info, edf_signals) => summary)
+    end
+    length(ambiguous) < 2 && return nothing
+    for ((info, edf_signals), (_, channel_name, _)) in ambiguous
+        filter!(!=(channel_name), info.channels)
+        filter!(!=(edf_signal), edf_signals)
+    end
+    return map(last, ambiguous)
+end
```

Comment:
I haven't thought this through super carefully but I have a suspicion that unless you're very careful there's a risk of removing BOTH+ versions of the ambiguous extracted signal...or  maybe that's what you want?  Either way could you add a comment explaining?

Reply:
Yeah I'm removing all/both extracted channels, because I don't know which to pick.

I thought of checking whether one of the signal type strings ("emg", "eeg" etc) appears anywhere in the label or transducer_type, then pick that one. Not sure how much of the wild data that would cover / whether that's worth the trouble.